examples/dreamer/dreamer.py

```diff
@@ -0,0 +1,355 @@
+    #     policy,
+    #     annealing_num_steps=cfg.annealing_frames,
+    #     sigma=cfg.ou_sigma,
+    #     theta=cfg.ou_theta,
+    # ).to(device)
+
+    action_dim_gsde, state_dim_gsde = None, None
+    proof_env.close()
+    create_env_fn = parallel_env_constructor(
+        cfg=cfg,
+        stats=stats,
+        action_dim_gsde=action_dim_gsde,
+        state_dim_gsde=state_dim_gsde,
+    )
+
+    collector = make_collector_offpolicy(
+        make_env=create_env_fn,
+        actor_model_explore=model_explore,
+        cfg=cfg,
+        # make_env_kwargs=[
+        #     {"device": device}
+        #     for device in cfg.collector_devices
+        # ],
+    )
+
+    replay_buffer = make_replay_buffer(device, cfg)
+
+    recorder = transformed_env_constructor(
+        cfg,
+        video_tag=video_tag,
+        norm_obs_only=True,
+        stats=stats,
+        logger=logger,
+    )()
+
+    # remove video recorder from recorder to have matching state_dict keys
+    if cfg.record_video:
+        recorder_rm = TransformedEnv(recorder.base_env)
+        for transform in recorder.transform:
+            if not isinstance(transform, VideoRecorder):
+                recorder_rm.append_transform(transform)
+    else:
+        recorder_rm = recorder
+
+    if isinstance(create_env_fn, ParallelEnv):
+        recorder_rm.load_state_dict(create_env_fn.state_dict()["worker0"])
+        create_env_fn.close()
+    elif isinstance(create_env_fn, EnvCreator):
+        recorder_rm.load_state_dict(create_env_fn().state_dict())
+    else:
+        recorder_rm.load_state_dict(create_env_fn.state_dict())
+    # reset reward scaling
+    for t in recorder.transform:
+        if isinstance(t, RewardScaling):
+            t.scale.fill_(1.0)
+            t.loc.fill_(0.0)
+
+    record = Recorder(
+        record_frames=cfg.record_frames,
+        frame_skip=cfg.frame_skip,
+        policy_exploration=policy,
+        recorder=recorder,
+        record_interval=cfg.record_interval,
+        log_keys=cfg.recorder_log_keys,
+    )
+
+    final_seed = collector.set_seed(cfg.seed)
+    print(f"init seed: {cfg.seed}, final seed: {final_seed}")
+    ## Training loop
+    collected_frames = 0
+    pbar = tqdm.tqdm(total=cfg.total_frames)
+    r0 = None
+    path = Path("./log")
+    path.mkdir(exist_ok=True)
+
+    scaler = GradScaler()
+    torch.cuda.empty_cache()
+    for i, tensordict in enumerate(collector):
+
+        # update weights of the inference policy
+        collector.update_policy_weights_()
+
+        if r0 is None:
+            r0 = tensordict["reward"].mean().item()
+        pbar.update(tensordict.numel())
+        if "mask" in tensordict.keys():
+            current_frames = tensordict["mask"].sum()
+            tensordict = tensordict[tensordict.get("mask").squeeze(-1)]
+        else:
+            current_frames = tensordict.numel()
+            collected_frames += current_frames
+        tensordict = tensordict.reshape(-1, cfg.batch_length)
```

Comment:
Why do we need this? Doesn't the tensordict already have this size?
This will break if there is a "mask" key in the tensordict (see my comment above)
Also reshape should be replaced by 'view' no?

Reply:
No, the tensordict has the size of the collected batch which is n_workers x max_frames_per_traj with max_frames_per_traj=1000, but then we want to use tensors of size B x batch_length with batch_length = 50 in dreamer default

Reply:
are we sure that the resulting tensordict will always be properly shaped? Also that might break no, if batch_length does not match the size of the collected data. e.g. what happens if max_frames_per_traj=789 and batch_length=25?